quint/src/runtime/testing.ts

```diff
@@ -74,22 +74,31 @@ export interface TestResult {
  * @param lookupTable lookup table as produced by the parser
  * @param analysisOutput the maps produced by the static analysis
  * @param options misc test options
- * @returns the results of running the tests
+ * @returns the `right(results)` of running the tests if the tests could be
+ *    run, or `left(errors)` with any compilation or analysis errors that
+ *    prevent the tests from running
  */
 export function compileAndTest(
   compilationState: CompilationState,
   mainName: string,
   lookupTable: LookupTable,
   options: TestOptions
 ): Either<QuintError[], TestResult[]> {
-  const recorder = newTraceRecorder(options.verbosity, options.rng)
   const main = compilationState.modules.find(m => m.name === mainName)
   if (!main) {
     return left([{ code: 'QNT405', message: `Main module ${mainName} not found` }])
   }
 
+  const recorder = newTraceRecorder(options.verbosity, options.rng)
   const ctx = compile(compilationState, newEvaluationState(recorder), lookupTable, options.rng.next, main.declarations)
 
+  const ctxErrors = ctx.syntaxErrors.concat(ctx.compileErrors, ctx.analysisErrors)
+  if (ctxErrors.length > 0) {
+    // In principle, these errors should have been caught earlier.
+    // But if they did not, return immediately.
```

Comment:
I know this comment isn't new, but I'm skeptical that we'd have caught compile errors before trying to compile :thinking:

Reply:
True, I guess this only applies to the syntax errors and type checking errors, which would have prevented from moving on to this stage in the pipeline.